### DISCUSSION TITLE
[MA v2 beta] incorrect album artist detected?

### DISCUSSION THREAD

**Discussion by DutchJaFO** [music-assistant/support#1152] [Q&A]
I've noticed that in the album view a few albums where showing a wrong arist name underneath the album name/image.

In all situations so far this was a tagging issue where the album had set the 'album artist' to 'soundtrack'.

Fixing the tag did resolve it for me, but there is one odd thing in my library.

The artist 'Stray Cats' did not appear in the 'artist view' when I enable the 'album artist' filter.
I also have this in my Album view :
![image](https://user-images.githubusercontent.com/39138434/226171862-c116122d-1f0c-4449-be03-09da13e58f57.png)

Which also shows tracks that don't belong to the album :
![image](https://user-images.githubusercontent.com/39138434/226172026-97a7d8fb-3710-4c5c-b88b-d341fb5edc85.png)

'Monkey Business' is an album by 'the black eyed peas' and has nothing to do with 'stray cats'.
I had a similar situation with a different album (tracks from a different listed under the wrong album), which was in part a tagging issue (incomplete tags). However it wasn't until I did resolve itself until a 'new' install of the MA v2 beta.

**Comment by OzGav** [comment]
I think the final point you make is a known issue which is being tracked in the backlog under "Updated tags not picked up after rescan ?" 

I understand from the above that all the odd issues you have you found were ultimately a tagging problem but fixing the tags did not immediately fix the issue. If I understand you correctly then the backlog item will hopefully rectify this for you.

**Comment by DutchJaFO** [comment]
that is likely to be related to that.
It probably will also fix the images not being detected after removing them.

I also suspect that some kind of tagging issue is causing the stray cats album to be missing from the album list, whereas 'the black eyed peas' are missing in the 'artist view' with 'album artist only'  filter active.

I'm currently at MA v2 beta 9.

Another possible relation exists with "Artwork in folder not picked up correctly" in the backlog

**Reply by OzGav** [reply]
Have you fully tagged those two albums with Picard?

**Reply by DutchJaFO** [reply]
both are fully tagged in Picard.

There are still albums with incomlete tags (usually no music brainz id's) as well as some with 'soundtrack' as album artist name in my smb-provider shares. I am in the process of fixing the latter.

I did remove the one track that was causing an exception in MA due to it missing one or essential tags for MA (it's the 'casba' track by 'abney park' that I shared with Marcel on his google drive in discord.

**Reply by OzGav** [reply]
Sounds good. Think we will have to wait until a fix comes in for the tag rescan. I raised that but for more minor issues than what you are seeing so hopefully it will fix it for both of us. You could try deleting from the database and the load it back in by navigating to the album via BROWSE and see if that makes any difference

**Reply by DutchJaFO** [reply]
I've deleted both artists and the album with the wrong tracks from the database.
After that I've navigated to 'browse' and could locate both artists and albums in there.
I noticed that the 'black eyed peas' where not showing images for all of the tracks, but 'stray cats' was.

I have now started a rescan of the smb-provider to see if that changes anything (it will take a while as this is the 'artist' folder ...)

**Reply by OzGav** [reply]
I have a related issue for track images not showing but that is when I have the album artist as "Various Artists". All I can suggest is just check that album is tagged correctly, I assume with "Black Eyed Peas" as the album artist

**Reply by DutchJaFO** [reply]
that worked for the albums ... both are back in album view
both artists are in the 'artist' view with 'album artist' enabled

However for some reason 'stray cats' are now listed under 'john waite' ... despite the album artist being 'stray cats' ???
ah ... it looks like this is another album that I need to fix as 'john waite' is on one of the GTA soundtrack albums.

I think this confirms that tags are not updated correctly after a re-sync after they are changed in the files.

**Comment by DutchJaFO** [comment]
Current work-around :
= fix tags in files 
= in MA : click 'remove from database' for artist/album affected by this
= go to 'browse' and check artist/albums
= re-sync the provider(s)

This fixed the monkey business/stray cats issue in my situation, but it also highlighted yet another album in need of re-tagging.

**Reply by OzGav** [reply]
Let me try that for a small change I made. I did all but the last step I think. Yeah when MA v1 was in its early stages I worked with Marcel to sort a bunch of issues but it highlighted lots of tagging issues that I had

**Reply by DutchJaFO** [reply]
yeah, when I started with MA v1 I did solve a lot of tagging issues as I had to re-tag my entire library.
I never checked again since adding Spotify to MA v1.

As I'm focussing on the smb-provider in MA v2 beta I'm discovering more tagging issues that need fixing.
It does reduce the amount of tests I can do because I'm not undoing that tagging ;)

**Reply by OzGav** [reply]
That sounds good. I am doing the same and have only added my one SMB provider at the moment. Yeah exactly, you can see why we just tell people to just tag things properly and then lots of automagic will then work.

**Reply by DutchJaFO** [reply]
I did a fresh install of v2 b9 and 'stray cats' dissappeared again and they're also back to 'monkey business'.
Something fishy is going on.
I was tagging a bunch of other albums during the scan ... so maybe that is a problem ?

I might have to stop my Plex and Emby servers from interfering with the tagging as well.
I hadn't thought of that as a potential source of issues 'til now. They shouldn't be causing problems, but they might be.

**Reply by OzGav** [reply]
I don't use those so can't help out there. I would say that changing tags during the scan could cause an issue although I would have thought you would see an error if MA couldn't scan a file.

**Reply by OzGav** [reply]
This should all be fixed in the latest betas
